src/interfaces/IAloeVolatilityOracleAdapter.sol

```diff
@@ -0,0 +1,144 @@
+
+    /**
+     * ////////// HELPERS ///////////
+     */
+
+    /**
+     * @notice Retrieves the uniswap v3 pool for passed ERC20 address plus arguments
+     * from setUniswapV3Pool.
+     * @param tokenA The contract address of the ERC20 for which to retrieve the v3 pool.
+     * @param tokenB The contract address of the ERC20 for which to retrieve the v3 pool.
+     * @param fee The fee tier for the pool in 1/100ths of a bip.
+     * @return pool The uniswap v3 pool for the supplied token.
+     */
+    function getV3PoolForTokensAndFee(address tokenA, address tokenB, uint24 fee)
+        external
+        view
+        returns (IUniswapV3Pool pool);
+
+    /**
+     * @notice Retrieves the uniswap fee in 1/100ths of a bip.
+     * @param tier The fee tier enum.
+     * @return The fee in 1/100ths of a bip.
+     */
+    function getUniswapV3FeeInHundredthsOfBip(UniswapV3FeeTier tier) external pure returns (uint24);
+
+    /**
+     * @notice Updates the cached implied volatility for the tokens in the refresh list.
+     * @return timestamp The timestamp of the cache refresh.
+     */
+    function refreshVolatilityCache() external returns (uint256 timestamp);
+
+    /**
+     * @notice Updates the cached implied volatility for the supplied pool info.
+     * @return timestamp The timestamp of the cache refresh.
+     */
+    function refreshVolatilityCacheAndMetadataForPool(UniswapV3PoolInfo calldata info)
+        external
+        returns (uint256 timestamp);
+
+    /**
+     * ////////// TOKEN REFRESH LIST //////////
+     */
+
+    /**
+     * @notice Sets the list of tokens and fees to periodically refresh for implied volatility.
+     * @param list The token refresh list.
+     * @return The token refresh list.
+     */
+    function setTokenFeeTierRefreshList(UniswapV3PoolInfo[] calldata list)
+        external
+        returns (UniswapV3PoolInfo[] memory);
+
+    /**
+     * @notice Gets the list of tokens and fees to periodically refresh for implied volatility.
+     * @return The token refresh list.
+     */
+    function getTokenFeeTierRefreshList() external view returns (UniswapV3PoolInfo[] memory);
+
+    /// function addTokenToRefreshList(address token) external returns (address);
+
+    /**
+     * /////////////// ADMIN FUNCTIONS ///////////////
+     */
+
+    /**
+     * @notice Sets the uniswap v3 factory contract address.
+     * @dev Used to get addresses of uniswap v3 pools for specific tokens.
+     * @param factory The address of the uniswap v3 factory contract.
+     * @return The address of the uniswap v3 factory contract.
+     */
+    function setV3Factory(address factory) external returns (address);
+
+    /**
+     * @notice Sets the aloe voltaility oracle contract address.
+     * @param oracle The contract address for the aloe volatility oracle.
+     * @return The contract address for the aloe volatility oracle.
+     */
+    function setAloeOracle(address oracle) external returns (address);
```

Comment:
Confused on the self referential storage setter here.

Reply:
The adapter wraps the aloe volatility oracle contract. This is exposed in case we want to change the contract address of the aloe oracle.